### PR TITLE
VKCI-281,282: Simplify VM creation and ExtraConfig updates

### PR DESCRIPTION
**Pull request by arunmk**
1. VM creation was nested and using redundant APIs to get the same information. 
2. VM creation had some complex logic around multi-VM creation and power-on. Those were added by me and the intern a long time ago, but they are not relevant now. They also hinder readability and ability to maintain and add new code.
3. ExtraConfig update updates one config at a time and each update is slow. They can be batched. (For a single-config update, one can use a single-element map).
4. There were some functions that were blocking with the task. Now these functions are essentially asynchronous with the task being returned to the caller. It is a simple pattern to block in the caller.
5. Some basic decomposition into functions to help readability.

All of the changes are in the common core. They affect caller codepaths on CSE and CAPVCD. Those will be handled by commits in those projects separately.

All of the above are tested by a small test written in the PR.

- - -
This change is [Reviewable](https://reviewable.io/reviews/vmware/cloud-provider-for-cloud-director/333)

### CODE REVIEW

**Review by Anirudh9794 — APPROVED** [review]
:lgtm: 

> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/vmware/cloud-provider-for-cloud-director/333#-:-NnpgkgICMSop1w_czcy:bg7igkv)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @rocknes and @sahithi)

File: pkg/vcdsdk/vapp.go [hunk 1]
```diff
@@ -421,16 +419,22 @@ func (vdc *VdcManager) getTaskFromResponse(resp *http.Response) (*govcd.Task, er
 	return task, nil
 }
 
-func (vdc *VdcManager) SetVmExtraConfigKeyValue(vm *govcd.VM, key string, value string, required bool) error {
+func (vdc *VdcManager) SetMultiVmExtraConfigKeyValuePairs(vm *govcd.VM, extraConfigMap map[string]string,
+	required bool) (govcd.Task, error) {
 	_, extraConfigVm, err := vdc.getVmExtraConfigs(vm)
 	if err != nil {
-		return fmt.Errorf("error retrieving vm extra configs: [%v]", err)
+		return govcd.Task{}, fmt.Errorf("error retrieving vm extra configs: [%v]", err)
```
Comment (rocknes) [inline pkg/vcdsdk/vapp.go:426]: Wouldn't sending nil be better?

Is there an easy way for the caller to know that this is an empty task object?

**Review by rocknes — APPROVED** [review]
:lgtm:

File: pkg/vcdsdk/vapp.go (hunk 1 above)
Comment (arunmk) [reply pkg/vcdsdk/vapp.go:426]: I am following the pattern in govcd code. I also found this as odd. However it is better to follow the pattern IMO.

**Pull request merged**